torch/csrc/autograd/profiler.h

```diff
@@ -188,6 +191,8 @@ struct RangeEventList {
                 "num_block_elements is calculated incorrectly");
   using block_type = std::vector<Event>;
 
+// allocBlock() assumes that mutex_ is held when called, in order to prevent
+  // multiple threads' block writes stomping over each other.
   void allocBlock() {
```

Comment:
if allocBlock is only called in record, and we assume that lock is held, shall we make it in private?